From 3837f9950a654b7c99389aff1f1ca805d5660526 Mon Sep 17 00:00:00 2001
From: Michael Pauzner <mpauzner@habana.ai>
Date: Tue, 02 Jan 2024 11:35:00 +0200
Subject: [PATCH] [FSW-1764] boot_desc: include added

Header file hl_fw_if.h has split in two files.
Thus, the include is updated or added.

Change-Id: Ic26fd04eac52d16077896519173bee040c3f30d2
Signed-off-by: Michael Pauzner <mpauzner@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index bb83ae8..e5f9a62 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -10,7 +10,7 @@
 #include <bitops.h>
 #include <baremetal.h>
 #include <device.h>
-#include <hl_fw_if.h> /* requires baremetal.h for __packed */
+#include <hl_fw_upd_if.h> /* requires baremetal.h for __packed */
 
 #include "common.h"
 #include "uart.h"
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 2ccf56a..a2695ef 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -25,7 +25,7 @@
 #include <gaudi2/gaudi2_security.h>
 #include <preboot/fit.h>
 #include <preboot/fit_rsa.h>
-#include <hl_fw_if.h>
+#include <hl_fw_upd_if.h>
 #include <psoc_arc0_mstr_if_core_hbw_regs.h>
 #include <psoc_arc0_mstr_if_core_lbw_regs.h>
 #include <psoc_arc1_mstr_if_core_hbw_regs.h>
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 865fb0d..11cf902 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -64,7 +64,7 @@
 #include <pci.h>
 #include <pif_ts.h>
 #include <hl_pif_arc_if.h>
-#include <hl_fw_if.h>
+#include <hl_fw_upd_if.h>
 #include <gaudi3_efuse.h>
 
 #define PIF_ARC_NUM_OF_MEM	2
diff --git a/include/boot_desc.h b/include/boot_desc.h
index 0e38a29..8cc69ab 100644
--- a/include/boot_desc.h
+++ b/include/boot_desc.h
@@ -9,6 +9,7 @@
 
 #include <baremetal.h>
 #include <hl_fw_if.h>
+#include <hl_fw_upd_if.h>
 #include <hl_boot_desc.h>
 #include <sha.h>
 
diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index b7d40f9..6507868 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -9,7 +9,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <baremetal.h>
-#include <hl_fw_if.h>		/* requires stdint.h */
+#include <hl_fw_upd_if.h>		/* requires stdint.h */
 #include <hl_boot_desc.h>
 #include <common.h>
 #include <hl.h>
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 31b1e61..3fc582a 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -13,6 +13,7 @@
 #include <baremetal.h>
 #include <device.h>
 #include <hl_fw_if.h>
+#include <hl_fw_upd_if.h>
 #include <hl_fit_if.h>
 #include <fit_load_type.h>
 #include <security.h>
